WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs

```diff
@@ -0,0 +1,214 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reactive.Linq;
+using Avalonia.Controls;
+using Avalonia.Controls.Models.TreeDataGrid;
+using Avalonia.Controls.Templates;
+using DynamicData;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Fluent.TreeDataGrid;
+using WalletWasabi.Fluent.ViewModels.CoinControl.Core;
+using WalletWasabi.Fluent.ViewModels.Wallets;
+using WalletWasabi.Fluent.Views.CoinControl.Core.Cells;
+using WalletWasabi.Fluent.Views.CoinControl.Core.Headers;
+
+namespace WalletWasabi.Fluent.ViewModels.CoinControl;
+
+public partial class CoinSelectorViewModel : ViewModelBase, IDisposable
+{
+	[AutoNotify] private IReadOnlyCollection<SmartCoin>? _selectedCoins;
+
+	public CoinSelectorViewModel(WalletViewModelBase walletViewModel, IEnumerable<SmartCoin> initialCoinSelection)
+	{
+		var pockets = walletViewModel.Wallet.GetPockets();
+		var pocketItems = CreatePocketItems(pockets);
+
+		SyncSelectedItems(pocketItems, initialCoinSelection);
+
+		var pocketColumn = PocketColumn();
+		TreeDataGridSource = new HierarchicalTreeDataGridSource<CoinControlItemViewModelBase>(pocketItems)
+		{
+			Columns =
+			{
+				ChildrenColumn(),
+				IndicatorsColumn(),
+				AmountColumn(),
+				AnonymityScoreColumn(),
+				pocketColumn,
+			}
+		};
+
+		var coins = GetAllCoinItems(pocketItems);
+
+		SelectedCoinsChanged = coins
+			.AsObservableChangeSet(x => x.SmartCoin)
+			.AutoRefresh(x => x.IsSelected, TimeSpan.FromMilliseconds(100), scheduler: RxApp.MainThreadScheduler)
+			.ToCollection()
+			.Select(x => x.Where(m => m.IsSelected == true))
+			.Select(models => models.Select(x => x.SmartCoin).ToImmutableList())
+			.ReplayLastActive();
+
+		SelectedCoinsChanged.BindTo(this, model => model.SelectedCoins);
+	}
+
+	public IObservable<IReadOnlyCollection<SmartCoin>> SelectedCoinsChanged { get; }
```

Comment:
Why is this still needed? Things can observe `SelectedCoins` with a `WhenAnyValue()`.
So this observable is unnecessary.